### ISSUE TITLE
Questions about FSRSDataset

### ISSUE THREAD

**Issue by dae** [open-spaced-repetition/fsrs-optimizer-burn#34]
@L-M-Sherlock I have a question about FSRSDataset before I refactor it. 

Currently there are ::train() and ::test() methods that return the same dataset. Is there a plan to make them return separate data in the future? Or would one method be enough?

Likewise with dataloader_train/_test - will they differ in the future? Or should they always have the same source data?

**Comment by L-M-Sherlock** [comment]
Test dataset will be empty in real optimization for users. FSRS will all data in the training.

**Comment by dae** [comment]
Sorry, I may need an ELI5 here :-) Are you saying that when users train their real data, we'll pass in the full vec of FSRS items to dataloader_train, and an empty vec to dataloader_test? What advantages are there in passing in the full data to dataloader_test when developing?

**Comment by L-M-Sherlock** [comment]
Yeah. We will pass an empty vec to dataloader_test. I haven't test that, so I passed full data to dataloader_test like the example provided in burn when developing. Please feel free to remove it.

**Comment by dae** [comment]
Sorry, one more question: is config.seed supposed to be random in actual use (outside of tests)?

**Comment by dae** [comment]
And one more :-) Is requestedRetention relevant for the training process, or something that only applies after the weights have been generated?

**Comment by L-M-Sherlock** [comment]
> Sorry, one more question: is config.seed supposed to be random in actual use (outside of tests)?

I recommend making it fixed. Because I don't want the user complain that the optimized parameters are always different over each training.

> Is requestedRetention relevant for the training process, or something that only applies after the weights have been generated?

The requestedRetention is unrelated for the training. Users should set it by themselves.
